### PR TITLE
mesa: add zink gallium driver

### PR DESCRIPTION
**Pull request by ardadem**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/mesa/template [hunk 1]
```diff
@@ -147,21 +147,27 @@ else
 	configure_args+=" -Dgallium-vdpau=disabled -Dgallium-va=disabled -Dgallium-xvmc=disabled"
 fi
 
-# empty values introduced by leading comma are not allowed; the whole enumeration can be empty
-# _gallium_drivers is not affected by this problem
-_vulkan_drivers=${_vulkan_drivers/=,/=}
-_dri_drivers=${_dri_drivers/=,/=}
-
-configure_args+=" ${_gallium_drivers} ${_vulkan_drivers} ${_dri_drivers}"
-
```
Comment (ericonr) [inline srcpkgs/mesa/template:156]: To minimize the diff, I'd leave this whole thing in and just remove `_gallium_drivers`

File: srcpkgs/mesa/template [hunk 2]
```diff
@@ -147,21 +147,27 @@ else
...
+	_gallium_drivers+=",zink"
+	makedepends+=" vulkan-loader"
+fi
+
+# empty values introduced by leading comma are not allowed; the whole enumeration can be empty
+# _gallium_drivers is not affected by this problem
+_vulkan_drivers=${_vulkan_drivers/=,/=}
+_dri_drivers=${_dri_drivers/=,/=}
+
+configure_args+=" ${_gallium_drivers} ${_vulkan_drivers} ${_dri_drivers}"
```
Comment (ericonr) [inline srcpkgs/mesa/template:169]: Then here have only

```suggestion
configure_args+=" ${_gallium_drivers}"
```

instead

**Review by ericonr — CHANGES_REQUESTED** [review]

File: srcpkgs/mesa/template (hunk 2 above)
Comment (ardadem) [reply srcpkgs/mesa/template:169]: Uh, it's visually better to keep drivers in same variable imo. So is it okay to keep configure_args same and move rest of the lines where it is?

Comment (ericonr) [reply srcpkgs/mesa/template:169]: If you make a lot of changes, it makes it harder when looking at the commit to know what's related and what isn't (and more importantly, if a given block is changing or not). That's why I'd prefer to keep them to a minimum.

I was looking at your diff and confused why cleaning up `_vulkan_drivers` was moved, when it had nothing to do with the PR changes.

**Comment by ericonr** [comment]
I'm slightly worried adding `vulkan-loader` here creates a cyclical dependency.

Will have to check later.

**Comment by ericonr** [comment]
Great, it doesn't create one.

Thanks!

**Pull request merged**